addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js

```diff
@@ -299,7 +299,7 @@ export class PaymentScreen extends Component {
         if (
             syncOrderResult &&
             syncOrderResult.length > 0 &&
-            this.currentOrder.wait_for_push_order()
+            syncOrderResult.some((order) => order.wait_for_push_order())
         ) {
             await this.postPushOrderResolve(syncOrderResult.map((res) => res.id));
```

Comment:
Should not only give the one that have `wait_for_push_order` true ?

Reply:
@davidmonnom yes, you are right